src/main.cpp

```diff
@@ -775,88 +775,110 @@ bool AcceptToMemoryPoolWorker(CTxMemPool &pool,
+            }
 
-        if (poolBytes < nLargestBlockSeen)
-        {
-            feeCutoff = std::max(feeCutoff, initFeeCutoff);
-            nFreeLimit = std::min(nFreeLimit, (double)nLimitFreeRelay);
-        }
-        else if (poolBytes < (nLargestBlockSeen * MAX_BLOCK_SIZE_MULTIPLIER))
-        {
-            // Gradually choke off what is considered a free transaction
-            feeCutoff =
-                std::max(feeCutoff, initFeeCutoff + ((maxFeeCutoff - initFeeCutoff) * (poolBytes - nLargestBlockSeen) /
-                                                        (nLargestBlockSeen * (MAX_BLOCK_SIZE_MULTIPLIER - 1))));
+            // Limit check. Make sure minlimterfee is not > maxlimiterfee
+            if (_dMinLimiterTxFee > _dMaxLimiterTxFee)
+            {
+                dMaxLimiterTxFee.value = dMinLimiterTxFee.value;
+                _dMaxLimiterTxFee = _dMinLimiterTxFee;
+            }
 
-            // Gradually choke off the nFreeLimit as well but leave at least DEFAULT_MIN_LIMITFREERELAY
-            // So that some free transactions can still get through
-            nFreeLimit = std::min(
-                nFreeLimit, ((double)nLimitFreeRelay - ((double)(nLimitFreeRelay - DEFAULT_MIN_LIMITFREERELAY) *
-                                                           (double)(poolBytes - nLargestBlockSeen) /
-                                                           (nLargestBlockSeen * (MAX_BLOCK_SIZE_MULTIPLIER - 1)))));
-            if (nFreeLimit < DEFAULT_MIN_LIMITFREERELAY)
-                nFreeLimit = DEFAULT_MIN_LIMITFREERELAY;
-        }
-        else
-        {
-            feeCutoff = maxFeeCutoff;
-            nFreeLimit = DEFAULT_MIN_LIMITFREERELAY;
-        }
+            // When the mempool starts falling use an exponentially decaying ~24 hour window:
+            // nFreeLimit = nFreeLimit + ((double)(DEFAULT_LIMIT_FREE_RELAY - nFreeLimit) / pow(1.0 - 1.0/86400,
+            // (double)(nNow - nLastTime)));
+            nFreeLimit /= std::pow(1.0 - 1.0 / 86400, (double)(nNow - nLastTime));
 
-        minRelayTxFee = CFeeRate(feeCutoff * 1000);
-        LOG(MEMPOOL,
-            "MempoolBytes:%d  LimitFreeRelay:%.5g  FeeCutOff:%.4g  FeesSatoshiPerByte:%.4g  TxBytes:%d  TxFees:%d\n",
-            poolBytes, nFreeLimit, ((double)::minRelayTxFee.GetFee(nSize)) / nSize, ((double)nFees) / nSize, nSize,
-            nFees);
-        if (fLimitFree && nFees < ::minRelayTxFee.GetFee(nSize))
-        {
-            static double dFreeCount = 0;
+            // When the mempool starts falling use an exponentially decaying ~24 hour window:
+            nMinRelay *= std::pow(1.0 - 1.0 / 86400, (double)(nNow - nLastTime));
 
-            // Use an exponentially decaying ~10-minute window:
-            dFreeCount *= std::pow(1.0 - 1.0 / 600.0, (double)(nNow - nLastTime));
-            nLastTime = nNow;
+            uint64_t nLargestBlockSeen = LargestBlockSeen();
+
+            if (poolBytes < nLargestBlockSeen)
```

Comment:
I think that this is a semantically equivalent simplification:
```nFreeLimit /= std::pow(1.0 - 1.0 / 86400, (double)(nNow - nLastTime));

// When the mempool starts falling use an exponentially decaying ~24 hour window:
nMinRelay *= std::pow(1.0 - 1.0 / 86400, (double)(nNow - nLastTime));

if (poolBytes < nLargestBlockSeen)
{
    nMinRelay = std::max(nMinRelay, _dMinLimiterTxFee);
    nFreeLimit = std::min(nFreeLimit, (double)nLimitFreeRelay);
}```

=>

```
if (poolBytes >= nLargestBlockSeen)
{
    nFreeLimit /= std::pow(1.0 - 1.0 / 86400, (double)(nNow - nLastTime));
    nMinRelay *= std::pow(1.0 - 1.0 / 86400, (double)(nNow - nLastTime));
}
```

Reply:
Needs to remain as is.  We need to calculate nFreelimit and nMinrelay on every iteration , since we need those values if we fall through to the other if else / else statements.